### PR TITLE
Better type inference in OscMessageBuilder.add_arg

### PR DESCRIPTION
**Pull request by Nagasaki45**

### CODE REVIEW

File: pythonosc/osc_message_builder.py [hunk 1]
```diff
@@ -78,6 +78,8 @@ def add_arg(self, arg_value, arg_type=None):
         arg_type = self.ARG_TYPE_TRUE
       elif builtin_type == builtins.bool and not arg_value:
         arg_type = self.ARG_TYPE_FALSE
+      else:
+        raise ValueError('Infered arg_value type is not supported')
```
Comment (attwad) [inline pythonosc/osc_message_builder.py]: The add_arg method can also now raise a TypeError if you use isinstance, can you add this to the documentation of this method please?

https://docs.python.org/3/library/functions.html#isinstance

File: pythonosc/test/test_osc_message_builder.py [hunk 2]
```diff
@@ -20,6 +20,10 @@ def test_wrong_param_raise(self):
     builder = osc_message_builder.OscMessageBuilder("")
     self.assertRaises(ValueError, builder.add_arg, "what?", 1)
 
+  def test_add_arg_invalid_infered_type(self):
+    builder = osc_message_builder.OscMessageBuilder('')
```
Comment (attwad) [inline pythonosc/test/test_osc_message_builder.py]: please keep the quotes style consistent and use ""

File: pythonosc/test/test_osc_message_builder.py [hunk 3]
```diff
@@ -20,6 +20,10 @@ def test_wrong_param_raise(self):
...
+    self.assertRaises(ValueError, builder.add_arg, {'name': 'John'})
```
Comment (attwad) [inline pythonosc/test/test_osc_message_builder.py]: here too please

File: pythonosc/test/test_osc_message_builder.py (hunk 2 above)
Comment (attwad) [inline pythonosc/test/test_osc_message_builder.py]: oh well... I just noticed this file is a mess in terms of consistency.. please ignore I'll fix all of that later on

File: pythonosc/osc_message_builder.py (hunk 1 above)
Comment (Nagasaki45) [inline pythonosc/osc_message_builder.py]: I disagree. `isinstance` will raise a `TypeError` only if the 2nd argument is not a type. Here, the 2nd argument is hardcoded with `str`, `bytes`, `int`, or `float` - all of them are proper python types.

**Comment by Nagasaki45** [comment]
Hi @attwad, please let me know if you want me to update/fix anything in this PR. Thanks!

**Comment by attwad** [comment]
All's good, sorry I was out today, merging this now!

**Comment by Nagasaki45** [comment]
Hi @attwad, can you please push a new version containing this PR to PyPI?

**Comment by attwad** [comment]
sure thing, done.

**Comment by Nagasaki45** [comment]
Wow, that was fast!
Thank you so much

**Pull request merged**